### ISSUE TITLE
Should `one_call_pipe_linter()` apply only to `%>%`?

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#2330]
I can see how `as.character(x)` can be preferable over `x %>% as.character()`, but I don't think the alternatives for other pipes work as well.

Consider `%<>%` (e.g.)

This example lints:

``` r
library(lintr)
lint(text = "data %<>% mutate(n.obs = n)", linters = one_call_pipe_linter())
#> <text>:1:2: warning: [one_call_pipe_linter] Expressions with only a single call shouldn't use pipe %<>%.
#>  data %<>% mutate(n.obs = n)
#>  ^~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<sup>Created on 2023-11-21 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

To avoid this, I instead need to do `data <- mutate(data, n.obs = n)`, and avoiding repetition of `data` is the entire _raison d'etre_ of `%<>%`. So when `{lintr}` says one shouldn't use this pipe only with a single call, it's basically saying don't use this pipe for the very thing for which it was built. That's not the case for `%>%`/`|>`. I hope I am being cogent here.

I could imagine a similar argument for other pipes. 

---

So I am wondering if the following should be changed only to cover `%>%`/`|>`:

https://github.com/r-lib/lintr/blob/209e250d499630cd6a5aab7319696d725e4b81b1/R/one_call_pipe_linter.R#L48

**Comment by MichaelChirico** [comment]
Hmm I think it makes sense to drop `%<>%`, not sure about the other pipes. E.g. `%$%`

```r
mtcars %$% cyl
```

Should really be `mtcars$cyl`.

And `%T>%`, there's really no need to use it unless it's part of a longer pipeline AIUI.

`%!>%` is identical to `%>%` for the purposes here.

So it's really only `%<>%` to worry about.

(Of course, the style guide also prohibits `%<>%`, but your point about `%<>%` + `one_call_pipe_linter()` applies regardless of whether a user adheres to the style guide)

**Comment by IndrajeetPatil** [comment]
Okay, fair enough. We can just carve out an exception for `%<>%` then.
